### PR TITLE
chain/bitcoind: connect to zmq in Start() to return error

### PR DESCRIPTION
**Pull request by cfromknecht**
Simple change that attempts to connect to bitcoind's zmq socket during the `Start()` method, so that any errors can be propagated back to the caller. Before the `socketHandler` could fail to connect and then exit, which could lead to unexpected hanging of the backend.

This was discovered because my bitciond had not been compiled with zmq headers, which caused the connection to error out and stall the lnd unit tests. I've tested this change again against bitciond with `--disable-zmq` to verify that the failure to connect error is returned during the backend's setup.

### CODE REVIEW

File: chain/bitcoind.go [hunk 1]
```diff
@@ -359,13 +359,21 @@ func (c *BitcoindClient) Start() error {
 		return errors.New("mismatched networks")
 	}
 
+	// Connect a ZMQ socket for block notifications
+	zmqClient, err := gozmq.Subscribe(c.zmqConnect, []string{"rawblock",
+		"rawtx"}, c.zmqPollInterval)
+	if err != nil {
+		log.Error(err)
```
Comment (aakselrod) [inline chain/bitcoind.go]: This bit is probably unnecessary since the error is being returned, but it doesn't hurt.

**Review by aakselrod — APPROVED** [review]
LGTM. One small suggestion about redundant logging.

**Comment by cfromknecht** [comment]
Good suggestion on redundant logging, all gtg!

**Review by Roasbeef — APPROVED** [review]
LGTM 💈

**Pull request merged**